storage/pruner/pruner.go

```diff
@@ -0,0 +1,374 @@
+package pruner
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/storage"
+	pebbleStorage "github.com/onflow/flow-go/storage/pebble"
+)
+
+type PruneMethod int
+
+const (
+	// PruneByHeight prunes data using finalized block heights
+	// This is should be used when a db does not include a view index
+	PruneByHeight PruneMethod = iota + 1
+
+	// PruneByView prunes data using views
+	// This is the simpler and more efficient method, but requires the node to have a view index
+	PruneByView
+
+	// DefaultPruneInterval is the default interval between pruning cycles
+	DefaultPruneInterval = 5 * time.Minute
+
+	// DefaultThrottleDelay is the default delay between pruning individual blocks
+	DefaultThrottleDelay = 20 * time.Millisecond
+)
+
+type Pruner struct {
+	component.Component
+
+	log zerolog.Logger
+	db  *pebble.DB
+
+	headers        storage.Headers
+	chunkDataPacks storage.ChunkDataPacks
+
+	targetKeepBlocks uint64 // number of blocks to keep below the highest block
```

Comment:
let's be more specific
```suggestion
	targetKeepBlocks uint64 // number of blocks to keep below the highest executed and sealed block
```